plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/internal/DeployArtifactValidator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.deploy.ui.internal;
+
+import com.google.cloud.tools.eclipse.appengine.deploy.ui.Messages;
+import com.google.common.base.Preconditions;
+import java.io.File;
+import org.eclipse.core.databinding.observable.value.IObservableValue;
+import org.eclipse.core.databinding.validation.ValidationStatus;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+
+public class DeployArtifactValidator extends FixedMultiValidator {
+
+  private final IPath basePath;
+  private final IObservableValue deployArtifactPath;
+
+  public DeployArtifactValidator(IPath basePath, IObservableValue deployArtifactPath) {
+    Preconditions.checkArgument(basePath.isAbsolute(), "basePath is not absolute.");
+    Preconditions.checkArgument(String.class.equals(deployArtifactPath.getValueType()));
+    this.basePath = basePath;
+    this.deployArtifactPath = deployArtifactPath;
+  }
+
+  @Override
+  protected IStatus validate() {
+    String pathValue = deployArtifactPath.getValue().toString();
+    if (pathValue.isEmpty()) {
+      return ValidationStatus.error(Messages.getString("error.deploy.artifact.empty"));
+    } else if (!pathValue.endsWith(".war") && !pathValue.endsWith(".jar")) {
```

Comment:
Cab we rely on the extension to tell us the type?

Reply:
Not really unless we unpack and examine archives, but this is a minimum effort. (I expect WARs are always ".war" and JARs ".jar".) We could add a more heavy checker later, which will examine a lot of internal contents of WAR/JAR.

Reply:
BTW, we don't need to determine if it's a WAR or a JAR in CT4E. We just stage whatever archive is there, and `gcloud app deploy` will pick a right runtime (the Jetty runtime or the plain Java runtime) based on the archive in the stage directory.